java-manta-client/src/main/java/com/joyent/manta/client/multipart/EncryptedMultipartManager.java

```diff
@@ -277,9 +285,15 @@ public ByteArrayOutputStream call(final long uploadedBytes) throws IOException {
                             }
                         }
                     });
+            final MantaMultipartUploadPart part = wrapped.uploadPart(upload.getWrapped(), partNumber, entity);
             encryptionState.setLastPartNumber(partNumber);
-            return wrapped.uploadPart(upload.getWrapped(), partNumber, entity);
+            return part;
         } finally {
+            if (encryptionState.getLastPartNumber() != partNumber && snapshot != null) {
```

Comment:
Is the rewind done for *any* error during `wrapped.uploadPart`?  Previously the part ordering validation effectively kept consumers from shooting themselves in the foot with retrying parts.  I'm trying to figure out of this opes up any new error paths.